### PR TITLE
Initialization refactor

### PR DESCRIPTION
**Pull request by danbarla**
This PR refactors the initialization of the factor graphs.
All functions which were in initialize_solution_utils.cpp are now part of Initializer class which is given as input to the trajectory.
Should make life easier when different types of initializations are needed.

### CODE REVIEW

File: gtdynamics/utils/ChainInitializer.cpp [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+ * @file  ChainInitializer.h
+ * @brief Utility methods for initializing trajectory optimization solutions for chain graph.
+ * @author: Dan Barladeanu
+ */
+
+#include <gtdynamics/utils/ChainInitializer.h>
+
+namespace gtdynamics {
+
+/*Values Initializer::ZeroValues(const Robot& robot, const int t, double gaussian_noise,
```
Comment (dellaert) [inline gtdynamics/utils/ChainInitializer.cpp:18]: Kill commented out code. If you need to save some code, put it in a local branch

File: gtdynamics/utils/ChainInitializer.cpp [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    std::vector<DynamicsSymbol> keys = {TorqueKey(j, t), JointAngleKey(j, t),
+                                        JointVelKey(j, t), JointAccelKey(j, t)};
+
+    for (size_t i = 0; i < keys.size(); i++)  {
+      //if (i==3 and t==4) continue;
+      values.insert(keys[i], sampler.sample()[0]);
+    }
+  }
+
+  //if (contact_points) {
```
Comment (dellaert) [inline gtdynamics/utils/ChainInitializer.cpp:89]: ?

File: gtdynamics/utils/ChainInitializer.h [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+     *      Noise drawn from a zero-mean gaussian distribution with a standard
+     *      deviation of gaussian_noise.
+     * @param[in] contact_points Contact points for timestep t.
+     */
+    gtsam::Values ZeroValues(
+        const Robot& robot, const int t, double gaussian_noise = 0.0,
+        const boost::optional<PointOnLinks>& contact_points = boost::none) const override;
+};
+
+} // namespace gtdynamics
```
Comment (dellaert) [inline gtdynamics/utils/ChainInitializer.h:38]: missing newline at end of file

File: gtdynamics/utils/Initializer.h [hunk 4]
```diff
@@ -0,0 +1,223 @@
+/* ----------------------------------------------------------------------------
```
Comment (dellaert) [inline gtdynamics/utils/Initializer.h:1]: nit, would have been better if this was renamed and adapted from initialize_solutions.h, for reviewing/history.

**Review by dellaert — APPROVED** [review]
Awesome! IC fails, so you need to look into that before merging..

**Comment by varunagrawal** [comment]
FYI CI will be resolved once we merge in #348

File: gtdynamics/utils/ChainInitializer.cpp (hunk 1 above)
Comment (danbarla) [reply gtdynamics/utils/ChainInitializer.cpp]: done

File: gtdynamics/utils/ChainInitializer.cpp (hunk 2 above)
Comment (danbarla) [reply gtdynamics/utils/ChainInitializer.cpp]: done

File: gtdynamics/utils/ChainInitializer.h (hunk 3 above)
Comment (danbarla) [reply gtdynamics/utils/ChainInitializer.h]: done

**Comment by danbarla** [comment]
@dellaert ok to merge?

**Comment by dellaert** [comment]
Sure, merge at will!

File: gtdynamics/utils/Initializer.h [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+    // Default Constructor
+    Initializer() {}
+
+    /**
+     * Add zero-mean gaussian noise to a Pose3.
+     *
+     * @param T Pose3 which to add noise to.
+     * @param sampler Helper to sample values from a gaussian distribution.
+     */
+    gtsam::Pose3 AddGaussianNoiseToPose(const gtsam::Pose3& T,
```
Comment (varunagrawal) [inline gtdynamics/utils/Initializer.h:44]: Apologies for bringing this up so late, but shouldn't the method names start with lower case if they are not static methods? This is to be consistent with the rest of the codebase (even though it is not exactly inline with the Google Style).

The [ISO C++ Standard](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#S-naming) is a better guide IMO.

**Pull request merged**